### DISCUSSION TITLE
Read `reject` from `.ncurc.json` to skip some dependencies

### DISCUSSION THREAD

**Discussion by DerZyklop** [renovatebot/renovate#24859] [Suggest an Idea]
### What would you like Renovate to be able to do?

Make renovate read the [ignoreDeps](https://docs.renovatebot.com/configuration-options/#ignoredeps) from property `reject` in **.ncurc.json** if available.

### Background 

There is a nice command-line tool called [npm-check-updates](https://www.npmjs.com/package/npm-check-updates) (also called `ncu`).
This tool lists dependencies that could be updated, their installed version, and there newest available version. 

![screenshot](https://user-images.githubusercontent.com/912344/178247906-e9706a52-0ccf-46d9-9e4a-20a293537c13.png)

Not important for this feature request, but just so you know: If one wants to update one of these, the tool can be used to run this update and write the new version to the **package.json**.

Even though we install renovate, we will continue to use `ncu` in some package.json scripts.

`ncu` provides a way to reject/ignore specific packages. Therefore, you need to define them in an array called `"reject"` inside the file **.ncurc.json**.
So, `"reject"` in **.ncurc.json** is basially the same as [ignoreDeps](https://docs.renovatebot.com/configuration-options/#ignoredeps) for **renovate.json**.

Using `ncu` and `renovate` in the same project leads to two ignore lists. One in `reject` in **.ncurc.json** and in probably almost all cases the exact same in `ignoreDeps` in **renovate.json**.

### If you have any ideas on how this should be implemented, please tell us here.

Add an option to **renovate.json** like `readNcuConfig`.

If `readNcuConfig` is set to `true`, read `reject` from **.ncurc.json** and handle it like `ignoreDeps` in **renovate.json**.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by MaronHatoum** [comment]
Hi @DerZyklop 
Thank you for all the information and the description.
I don't think that Renovate should depend on another config file and run it using this configuration(this is what I understood from the description above correct me if I'm wrong).

**Comment by HonkingGoose** [comment]
## Two places to ignore dependencies

> If `readNcuConfig` is set to `true`, read `reject` from **.ncurc.json** and handle it like `ignoreDeps` in **renovate.json**.

Hmm, that way we end up with two places where the ignored deps can live:

1. `ignoreDeps` in `renovate.json`
2. `reject` field in `.ncurc.json`

The above will get people confused about _where_ they're ignoring packages.

I think ignoring dependencies in the Renovate config file is easiest to reason about: "I'm using Renovate, so my ignored dependencies are in the Renovate config in `renovate.json`".

## How popular is `ncu`?

Even if having two places to ignore dependencies is confusing, it may still be worth adding support for the `reject` field in `.ncurc.json`. Do many people use `ncu`?
